server/repo.go

```diff
@@ -137,6 +138,19 @@ func PatchRepo(c *gin.Context) {
 		repo.Counter = *in.BuildCounter
 	}
 
+	if in.Name != nil && in.Owner != nil {
+		from, err := remote.Repo(user, *in.Owner, *in.Name)
+		if err != nil {
+			c.AbortWithError(http.StatusInternalServerError, err)
+		}
+		repo.Name = from.Name
+		repo.Owner = from.Owner
+		repo.FullName = from.FullName
+		repo.Avatar = from.Avatar
+		repo.Link = from.Link
+		repo.Clone = from.Clone
```

Comment:
Ok I'll totally add that.  Right now I have the CLI run a repair after the update which seems to work seamlessly from the user perspective. At least with what I've tried it works.